OpenRA.Mods.Common/Traits/Armament.cs

```diff
@@ -141,15 +147,20 @@ public Armament(Actor self, ArmamentInfo info)
 
 		public virtual WDist MaxRange()
 		{
-			return new WDist(Util.ApplyPercentageModifiers(Weapon.Range.Length, rangeModifiers));
+			return new WDist(Util.ApplyPercentageModifiers(Weapon.Range.Length, rangeModifiers.Select(m => m.GetRangeModifier())));
 		}
 
 		protected override void Created(Actor self)
 		{
 			turret = self.TraitsImplementing<Turreted>().FirstOrDefault(t => t.Name == Info.Turret);
 			ammoPool = self.TraitsImplementing<AmmoPool>().FirstOrDefault(la => la.Info.Name == Info.AmmoPoolName);
 			coords = self.Trait<BodyOrientation>();
-			rangeModifiers = self.TraitsImplementing<IRangeModifier>().ToArray().Select(m => m.GetRangeModifier());
+			notifyBurstComplete = self.TraitsImplementing<INotifyBurstComplete>().ToArray();
+			reloadModifiers = self.TraitsImplementing<IReloadModifier>().ToArray();
+			rangeModifiers = self.TraitsImplementing<IRangeModifier>().ToArray();
+			damageModifiers = self.TraitsImplementing<IFirepowerModifier>().ToArray();
+			inaccuracyModifiers = self.TraitsImplementing<IInaccuracyModifier>().ToArray();
+			notifyAttacks = self.TraitsImplementing<INotifyAttack>().ToArray();
```

Comment:
One thing to add to RoosterDragons explanation, be careful to but the select after the ToArray call and you will probably need to change the type of the members to IEnumerable as that is the type that Select returns.